### PR TITLE
win and tie logic added

### PR DESCRIPTION
**Pull request by s-furlong**
Game Should be playable for the DEMO with the tie and win logic

### CODE REVIEW

File: lib/game.rb [hunk 1]
```diff
@@ -57,10 +57,60 @@ def valid_move(index)
     end
   end
 
+  def horizontal_win
+    top_row_win = (@board.new_board[0] != '1') && three_all_same_token(@board.new_board[0, 3])
+    middle_row_win = (@board.new_board[3] != '4') && three_all_same_token(@board.new_board[3, 3])
+    bottom_row_win = (@board.new_board[6] != '7') && three_all_same_token(@board.new_board[6, 3])
+    if top_row_win || middle_row_win || bottom_row_win
```
Comment (twopoint718) [inline lib/game.rb]: The expression `top_row_win || middle_row_win || bottom_row_win` itself evaluates to true/false, so you can skip the if/else expression:

```ruby
def horizontal_win
  # ...
  top_row_win || middle_row_win || bottom_row_win
end
```

File: lib/game.rb [hunk 2]
```diff
@@ -57,10 +57,60 @@ def valid_move(index)
     end
   end
 
+  def horizontal_win
```
Comment (twopoint718) [inline lib/game.rb]: It's usually idiomatic to name methods like this (that just return a boolean) with a question mark, like this:

```ruby
def horizontal_win?
  # ...
```

**Review by twopoint718 — APPROVED** [review]
I mentioned some style changes that you may want to consider, but this looks good!

File: lib/game.rb (hunk 1 above)
Comment (s-furlong) [reply lib/game.rb]: this change has been updated to all my win logic

File: lib/game.rb (hunk 2 above)
Comment (s-furlong) [reply lib/game.rb]: this change has been added to all my win tie logic

**Pull request merged**